tests/cbtest/gtest/test_zrepl_prot.cc

```diff
@@ -571,3 +696,78 @@ TEST_F(ZreplDataTest, ReadBlockWithoutMeta) {
 		offset += sizeof (buf);
 	}
 }
+
+/*
+ * This test has many steps. If it proves to be too complicated, then split it
+ * into multiple smaller tests. It creates:
+ *
+ *   1 zvol with default target IP
+ *   1 zvol with explicit target IP
+ *    - restart zrepl -
+ *   1 zvol with default target IP
+ *   1 zvol with explicit target IP
+ *   destroy all zvols
+ *
+ * Verify that zrepl establishes and tears down connections as appropriate.
+ */
+TEST(TargetIPTest, CreateAndDestroy) {
```

Comment:
testcase of target restart and then reconnect from zrepl

Reply:
good point. I will add that.